src/AggregateFunctions/AggregateFunctionDeltaSumTimestamp.cpp

```diff
@@ -28,11 +28,11 @@ AggregateFunctionPtr createAggregateFunctionDeltaSumTimestamp(
         throw Exception("Incorrect number of arguments for aggregate function " + name,
             ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
-    if (!isInteger(arguments[0]) && !isFloat(arguments[0]) && !isDateOrDateTime(arguments[0]))
+    if (!isInteger(arguments[0]) && !isFloat(arguments[0]) && !isDate(arguments[0]) && !isDateTime(arguments[0]))
```

Comment:
I cannot understand the logic in this change.

Looks like it uncovered a huge tech debt after merging #7170,
and this tech debt has to be fixed ASAP.

Both `isDateTime` and `isDateOrDateTime` should either include or not include the `DateTime64` data type.

Reply:
And how exactly it is related to the error "AggregateFunctionFactory returned nullptr"?

Reply:
> Both isDateTime and isDateOrDateTime should either include or not include the DateTime64 data type.

We have separate methods for DateTime and DateTime64: 
https://github.com/ClickHouse/ClickHouse/blob/master/src/DataTypes/IDataType.h#L325-L326

I think checking DateTime64 in isDateOrDateTime method is not good, because DateTime64 column is implemented by ColumnDecimal, while Date and DateTime columns by ColumnVector. 

> And how exactly it is related to the error "AggregateFunctionFactory returned nullptr"?

nullptr is returned here:
https://github.com/ClickHouse/ClickHouse/blob/master/src/AggregateFunctions/Helpers.h#L263-L267

deltaSumTimestamp doesn't support working with DateTime64. I suppose author (and me too) didn't know that isDateOrDateTime return true for DateTime64.

Now I think that I should also add DateTime64 in https://github.com/ClickHouse/ClickHouse/blob/f22702001831c074c789be306a9f77c5406d848d/src/AggregateFunctions/Helpers.h#L249 and https://github.com/ClickHouse/ClickHouse/blob/f22702001831c074c789be306a9f77c5406d848d/src/AggregateFunctions/Helpers.h#L231 and set DateTime64Supported = false for deltaSumTimestamp

Reply:
Ok